sdk/healthinsights/azure-health-insights-clinicalmatching/src/test/java/com/azure/health/insights/clinicalmatching/ClinicalMatchingClientTest.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.health.insights.clinicalmatching;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import com.azure.health.insights.clinicalmatching.models.TrialMatcherResult;
+import com.azure.health.insights.clinicalmatching.models.TrialMatcherInference;
+import com.azure.health.insights.clinicalmatching.models.TrialMatcherPatientResult;
+import com.azure.core.util.serializer.TypeReference;
+
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Unit tests for {@link ClinicalMatchingClient}.
+ */
+public class ClinicalMatchingClientTest extends ClinicalMatchingClientTestBase {
+
+    private ClinicalMatchingClient getClient() {
+        return getClientBuilder().buildClient();
+    }
+
+    @Test
+    public void test() {
+        // playback
+        try {
+            testTMWithResponse(request -> {
+
+                BinaryData responseValue = getClient().beginMatchTrials(request, new RequestOptions()).waitForCompletion().getValue();
```

Comment:
https://github.com/Azure/azure-sdk-for-java/wiki/Cadl-Java-Quickstart#recordplayback-tests

Reply:
and why use the protocol method?

I assume you'd like to test the method that takes `TrialMatcherData`, instead of `BinaryData`?

Reply:
we are releasing public preview this week.
I used TrialMatcherData in the sample.
Very soon we will add more tests and more samples.